packages/headless/src/api/analytics/analytics.ts

```diff
@@ -115,13 +116,15 @@ interface ConfigureAnalyticsOptions {
   state: StateNeededByAnalyticsProvider;
   logger: Logger;
   analyticsClientMiddleware?: AnalyticsClientSendEventHook;
+  preprocessAnalyticsRequestMiddleware?: PreprocessRequestMiddleware;
```

Comment:
Seeing this new option next to `analyticsClientMiddleware` makes me question whether this is the best experience. But I also do not see a way to add the functionality to `analyticsClientMiddleware` that is non-breaking hmm,

Reply:
🤔  Another approach would have been to not have the `preprocessRequestMiddleware` under `search`, but more of a general function, used by all API/Services we use under the hood.

That could be breaking, or we could mark the old one as deprecated, still available, with a notice to use the new general function.
The signature could stay the same.

Reply:
Yeah having two different methods seems pretty confusing. Is the major difference is that the `analyticsClientMiddleware` lets you modify the payload before it's turned into a blob?

Reply:
`analyticsClientMiddleware` is for modifying the event payload according to the type, pretty basic and simple, we chain multiple hooks in the analytics library

`preprocessAnalyticsRequestMiddleware` lets you access the fetch/beacon request itself and modify it's options

both preprocess middlewares work a little differently, they don't have the same options, so we cannot really merge them without breaking.
I'm seeing these a more advanced use cases, I don't think we should deprecate them right now, let them evolve a bit and refactor them when we release a new version of headless with breaks

Reply:
👍